### DISCUSSION TITLE
more automation - auto-filling chanel tvg-ids by bot?

### DISCUSSION THREAD

**Discussion by Owyn** [iptv-org/community#5] [Ideas]
Maybe instead of using channels.csv and tediously matching channel IDs for each stream by hand - it could be possible to just make bot do it via a script? there's a "native_name" field in the channels.csv which should be the same as a channel official name and the one used when adding channels

even if a native name field isn't filled at all - we could just go fill it once (instead of placing tvg-ids by hand into added channels each time)

what do you think?

https://github.com/iptv-org/api

e.g. a user addes a stream:
```
#EXTINF:-1,Europa Plus TV
https://url...
```

bot checks channels.csv for such name and when he finds it 

`EuropaPlusTV.ru,Europa Plus TV,,,RU,,,c/RU,rus,music,FALSE,https://europaplustv.com/,`

bot later adds `tvg-id` into it from the found channel and it becomes
```
#EXTINF:-1 tvg-id="EuropaPlusTV.ru",Europa Plus TV
https://url...
```

in the end streams file

I know about chanels with the same name, but I think correct one could be found by first filtering by `country` or `broadcast_area` or `language`

additional automation ideas: 
- add `status=""` field by the bot too so user doesn't have it?
- add stream resolution by the bot too so user doesn't have it?
- for channels without a tvg-id but from a specific region - e.g. streams from China - should have language=CN added to them anyway even if there is no tvg-id on it?

**Comment by Owyn** [comment]
populating currently empty native_names in channels.csv can also be done by a script\bot by reverse matching - taking channel_ids from channels.csv and finding streams with such ids in streams file and taking channel names from there as native_names into channels.csv

**Comment by BellezaEmporium** [comment]
For this, bot should be able to notify us that a new channel with no info appeared, after being added in a list. It should be possible via a trigger of some sort, explaining it vulgarly could be : 
- a trigger placed on the playlists named "onPlaylistAddition" should look on @iptv-org/api 
- looks after "channels.csv"
- reads all the channels (it should normally be better to have a JSON, like what we have for developers, it simplifies view and we can create programs off it). 
- If it finds no occurences : it creates it and sends a message to us, telling that there's a new channel added that requires our attention (to fill it out)
- If it finds one, it's all good.

This means we require "tvg-id" to be filled before doing anything, or we automatize the tvg-id creation (either by taking the name of the channel, or translating it like we did before).
We also can use our @iptv-org/epg repo, to see if we have a channel name that looks like the one we added.

**Comment by freearhey** [comment]
Now to add a link, you just have to copy the ID from the [iptv-org.github.io](https://iptv-org.github.io/) and fill out the [form](https://github.com/iptv-org/iptv/issues/new?template=1_streams_add.yml), so in most cases you no longer have to edit files manually.
